modular_splurt/code/modules/mob/living/carbon/human/simple_animal/hostile/deathclaw/funclaw.dm

```diff
@@ -0,0 +1,180 @@
+			else
+				chosen_hole = CUM_TARGET_ANUS
+		if(2)
+			chosen_hole = CUM_TARGET_THROAT
+
+/mob/living/simple_animal/hostile/deathclaw/funclaw/proc/do_lewd_action(mob/living/M)
+	if(get_refraction_dif() > 0)
+		return
+
+	switch(chosen_hole)
+		if(CUM_TARGET_ANUS)
+			if(tearSlot(M, SLOT_WEAR_SUIT))
+				return
+			if(tearSlot(M, SLOT_W_UNIFORM))
+				return
+
+			// Abomination deathclaws do other stuff instead
+			if(deathclaw_mode == "abomination" && M.client.prefs.unholypref == "Yes")
+				if(prob(1))
+					do_grindmouth(M)
+				else
+					do_grindface(M)
+				handle_post_sex(25, null, M)
+			else
+				do_anal(M)
+
+		if(CUM_TARGET_VAGINA)
+			if(tearSlot(M, SLOT_WEAR_SUIT))
+				return
+			if(tearSlot(M, SLOT_W_UNIFORM))
+				return
+
+			// Abomination deathclaws do other stuff instead
+			if(deathclaw_mode == "abomination" && M.client.prefs.unholypref == "Yes")
+				do_footjob_v(M)
+				handle_post_sex(10, null, M)
+			else
+				do_vaginal(M)
+
+		if(CUM_TARGET_THROAT)
+			if(tearSlot(M, SLOT_HEAD))
+				return
+			if(tearSlot(M, SLOT_WEAR_MASK))
+				return
+
+			// Abomination deathclaws do other stuff instead
+			if(deathclaw_mode == "abomination" && M.client.prefs.unholypref == "Yes")
+				if(prob(1))
+					do_faceshit(M)
+				else
+					do_facefart(M)
+				handle_post_sex(25, null, M)
+				shake_camera(M, 6, 1)
+			else
+				do_throatfuck(M)
+
+/mob/living/simple_animal/hostile/deathclaw/funclaw/cum(mob/living/M)
+
+	if(get_refraction_dif() > 0)
+		return
+
+	var/message
+
+	if(!istype(M))
+		chosen_hole = null
+
+	switch(chosen_hole)
+		if(CUM_TARGET_THROAT)
+			if(M.has_mouth() && M.mouth_is_free())
+				message = "shoves their fat reptillian cock deep down \the [M]'s throat and cums."
+			else
+				message = "cums on \the [M]'s face."
+		if(CUM_TARGET_VAGINA)
+			if(M.is_bottomless() && M.has_vagina())
+				message = "rams its meaty cock into \the [M]'s pussy and fills it with sperm."
+			else
+				message = "cums on \the [M]'s belly."
+		if(CUM_TARGET_ANUS)
+			if(M.is_bottomless() && M.has_anus())
+				message = "hilts its knot into \the [M]'s ass and floods it with Deathclaw jizz."
+			else
+				message = "cums on \the [M]'s backside."
+		else
+			message = "cums on the floor!"
+
+	if(deathclaw_mode == "abomination" && M.client.prefs.unholypref == "Yes")
+		message = "cums all over [M]'s body"
+
+	playsound(loc, "modular_splurt/sound/lewd/deathclaw[rand(1, 2)].ogg", 70, 1, -1)
+	visible_message("<font color=purple><b>\The [src]</b> [message]</font>")
+	shake_camera(M, 6, 1)
+	set_is_fucking(null ,null)
+
+
+	refractory_period = world.time + rand(150, 250) // Sex cooldown
+	set_lust(100) // Nuts at 400
+
+	sleep(20)
+	playsound(loc, "modular_sand/sound/interactions/slap.ogg", 70, 1, -1)
+	visible_message("<span class='danger'>\The [src]</b> slaps \the [M] right on the ass!</span>", \
+			"<span class='userdanger'>\The [src]</b> slaps \the [M] right on the ass!</span>", null, COMBAT_MESSAGE_RANGE)
```

Comment:
I don't currently know if there's a workaround for a chain of procs disallowing sleep() being called, but this is what I'd do for now 
```suggestion
	addtimer(CALLBACK(src, .proc/finish_spank, M), 20)

/mob/living/simple_animal/hostile/deathclaw/funclaw/proc/finish_spank(mob/living/M)
	playsound(loc, "modular_sand/sound/interactions/slap.ogg", 70, 1, -1)
	visible_message("<span class='danger'>\The [src]</b> slaps \the [M] right on the ass!</span>", \
			"<span class='userdanger'>\The [src]</b> slaps \the [M] right on the ass!</span>", null, COMBAT_MESSAGE_RANGE)COMBAT_MESSAGE_RANGE)
```